src/clusterfuzz/_internal/fuzzing/corpus_manager.py

```diff
@@ -603,25 +360,228 @@ def get_regressions_corpus_gcs_url(self):
+        storage.sign_delete_url(corpus_element_url))
+
+  # TODO(metzman): Add config to skip doing this when not needed (e.g. fuzz task
+  # will not update regressions corpus).
+  upload_urls = storage.get_arbitrary_signed_upload_urls(
+      gcs_url, num_uploads=10000)
+  last_updated = storage.last_updated(_get_gcs_url(bucket_name, bucket_path))
+  corpus = uworker_msg_pb2.Corpus(
+      corpus_urls=corpus_urls,
+      upload_urls=upload_urls,
+      gcs_url=gcs_url,
+  )
+  if last_updated:
+    timestamp = timestamp_pb2.Timestamp()  # pylint: disable=no-member
+    timestamp.FromDatetime(last_updated)
+    corpus.last_updated_time.CopyFrom(timestamp)
+
+  return corpus
+
+
+def get_target_bucket_and_path(engine,
+                               project_qualified_target_name,
+                               quarantine=False):
+  """Gets target and bucket path for the corpus."""
+  engine = os.getenv('CORPUS_FUZZER_NAME_OVERRIDE', engine)
+  if quarantine:
+    sync_corpus_bucket_name = environment.get_value('QUARANTINE_BUCKET')
+  else:
+    sync_corpus_bucket_name = environment.get_value('CORPUS_BUCKET')
+  if not sync_corpus_bucket_name:
+    raise RuntimeError('No corpus bucket specified.')
+  return sync_corpus_bucket_name, f'/{engine}/{project_qualified_target_name}'
+
+
+def get_fuzz_target_corpus(engine,
+                           project_qualified_target_name,
+                           quarantine=False,
+                           include_regressions=False):
+  """Copies the corpus from gcs to disk. Can run on uworker."""
+  fuzz_target_corpus = uworker_msg_pb2.FuzzTargetCorpus()
+  bucket_name, bucket_path = get_target_bucket_and_path(
+      engine, project_qualified_target_name, quarantine)
+  corpus = get_proto_corpus(bucket_name, bucket_path)
+  fuzz_target_corpus.corpus.CopyFrom(corpus)
+
+  if include_regressions:
+    regressions_bucket_path = f'{bucket_path}{REGRESSIONS_GCS_PATH_SUFFIX}'
+    fuzz_target_corpus.regressions_corpus.CopyFrom(
+        get_proto_corpus(bucket_name, regressions_bucket_path))
+  return fuzz_target_corpus
+
+
+def get_regressions_signed_upload_url(engine, project_qualified_target_name):
+  bucket, path = get_target_bucket_and_path(engine,
+                                            project_qualified_target_name)
+  regression_url = _get_regressions_corpus_gcs_url(bucket, path)
+  return storage.get_arbitrary_signed_upload_url(regression_url)
+
+
+def _sync_corpus_to_disk(corpus, directory):
+  shell.create_directory(directory, create_intermediates=True)
+  results = storage.download_signed_urls(corpus.corpus_urls, directory)
+  for filepath, upload_url in results:
+    corpus.filenames_to_delete_urls_mapping[filepath] = (
+        corpus.corpus_urls[upload_url])
+
+  # TODO(metzman): Add timeout and tolerance for missing URLs.
+  return results.count(None) < MAX_SYNC_ERRORS
+
+
+def fuzz_target_corpus_sync_to_disk(fuzz_target_corpus, directory) -> bool:
```

Comment:
Rather than all these `fuzz_target_corpus_` prefixed functions, could we perhaps just create a lightweight Python class that wraps the proto and adds these as methods? I think that would greatly aid readability.